File: python-pytest/abcmouse/pages/landing_page.py
from abcmouse.base.page import Page


class LandingPage(Page):
    def __init__(self, driver, url):
        super().__init__(driver, url)

    def signup(self):
        locator = [
            'document',
            '.querySelector(\"body > route-view\")',
            '.shadowRoot.querySelector(\"#page-component\")',
            '.shadowRoot.querySelector(\"main-layout > header > home-header '
            '> authstate-context:nth-child(3) > signup-button\")']
        return self.js_locator(locator)

File: CSharpNunitSelenium/CSharpNunitSelenium/ABCMouse/Pages/LandingPage.cs
using CSharpNunitSelenium.ABCMouse.Base;
using OpenQA.Selenium;

namespace CSharpNunitSelenium.ABCMouse.Pages;

public class LandingPage : Page
{
    
    protected override string URL { get; set; } = "https://www.abcmouse.com/abc/";
    public IWebElement? SignUp => (IWebElement) _jsExecutor!.ExecuteScript(
        "return document.querySelector(\"body > route-view\")" +
        ".shadowRoot.querySelector(\"#page-component\")" +
        ".shadowRoot.querySelector" +
        "(\"main-layout > header > home-header > authstate-context:nth-child(3) > signup-button\")"
        );
    public LandingPage(IWebDriver driver) : base(driver)
    {
    }
    
}
